bin/propolis-server/src/lib/server.rs

```diff
@@ -953,6 +955,77 @@ async fn instance_issue_crucible_snapshot_request(
     Ok(HttpResponseOk(()))
 }
 
+/// Issues a volume_construction_request replace to a crucible backend.
+#[endpoint {
+    method = PUT,
+    path = "/instance/disk/{id}/vcr",
+}]
+async fn instance_issue_crucible_vcr_request(
+    rqctx: RequestContext<Arc<DropshotEndpointContext>>,
+    path_params: Path<api::VCRRequestPathParams>,
+    request: TypedBody<api::InstanceVCRReplace>,
+) -> Result<HttpResponseOk<()>, HttpError> {
+    let path_params = path_params.into_inner();
+    let request = request.into_inner();
+    let new_vcr_json = request.vcr_json;
+    let disk_name = request.name;
+    let log = rqctx.log.clone();
+
+    // Get the instance spec for storage backend from the disk name.  We use
+    // the VCR stored there to send to crucible along with the new VCR we want
+    // to replace it.
+    let vm_controller = rqctx.context().vm().await?;
+    let mut spec = vm_controller.instance_spec().await;
```

Comment:
I think we need a better story for mutating a VM this way, but don't want to hold up progress on the replacement workflow while we work out exactly what that should look like. #205 is tracking the general "change a VM's configuration at runtime" story, so I suggest we refer to that here, and I'll bump its priority up accordingly.

```suggestion

    // TODO(#205): Mutating a VM's configuration should be a first-class
    // operation in the VM controller that synchronizes with ongoing migrations
    // and other attempts to mutate the VM. For the time being, use the instance
    // spec lock to exclude other concurrent attempts to reconfigure this
    // backend.
    let mut spec = vm_controller.instance_spec().await;
```

Reply:
Got it, I finally figured out what github was showing me here.